frontend/resourceadm/components/ResourcePartyLists/ResourcePartyLists.tsx

```diff
@@ -0,0 +1,142 @@
+import { useGetPartyListsQuery } from 'resourceadm/hooks/queries/useGetPartyLists';
+import { StudioSpinner } from '@studio/components';
+import { useGetResourcePartyListsQuery } from 'resourceadm/hooks/queries/useGetResourcePartyLists';
+import { useAddResourcePartyListMutation } from 'resourceadm/hooks/mutations/useAddResourcePartyListMutation';
+import { useRemoveResourcePartyListMutation } from 'resourceadm/hooks/mutations/useRemoveResourcePartyListMutation';
+import { getResourcePageURL } from 'resourceadm/utils/urlUtils';
+import { NewPartyListModal } from '../NewPartyListModal/NewPartyListModal';
+import { Resource } from 'app-shared/types/ResourceAdm';
+
+interface ResourcePartyListsProps {
+  env: string;
+  resourceData: Resource;
+}
+
+export const ResourcePartyLists = ({
+  env,
+  resourceData,
+}: ResourcePartyListsProps): React.ReactNode => {
+  const { t } = useTranslation();
+
+  const { selectedContext } = useParams();
+  const repo = `${selectedContext}-resources`;
+  const createPartyListModalRef = useRef<HTMLDialogElement>(null);
+
+  const [selectedLists, setSelectedLists] = useState<string[]>([]);
+
+  const {
+    data: envListData,
+    isLoading: isLoadingEnvListData,
+    error: envListDataError,
+  } = useGetPartyListsQuery(selectedContext, env);
+  const {
+    data: connectedLists,
+    isLoading: isLoadingConnectedLists,
+    error: connectedListsError,
+  } = useGetResourcePartyListsQuery(selectedContext, resourceData.identifier, env);
+  const { mutate: addResourcePartyList } = useAddResourcePartyListMutation(
+    selectedContext,
+    resourceData.identifier,
+    env,
+  );
+  const { mutate: removeResourcePartyList } = useRemoveResourcePartyListMutation(
+    selectedContext,
+    resourceData.identifier,
+    env,
+  );
+
+  useEffect(() => {
+    if (connectedLists) {
+      setSelectedLists(connectedLists.map((x) => x.partyListIdentifier));
+    }
+  }, [connectedLists]);
+
+  const handleRemove = (listItemId: string) => {
+    setSelectedLists((old) => old.filter((y) => y !== listItemId));
+    removeResourcePartyList(listItemId);
+  };
+
+  const handleAdd = (listItemId: string) => {
+    addResourcePartyList(listItemId);
+    setSelectedLists((old) => [...old, listItemId]);
+  };
+
+  if (isLoadingEnvListData || isLoadingConnectedLists) {
+    return <StudioSpinner spinnerText={t('general.loading')} />;
+  }
+
+  if (envListDataError || connectedListsError) {
+    return <Alert severity='danger'>{t('resourceadm.listadmin_load_list_error')}</Alert>;
+  }
+
+  return (
+    <div className={classes.resourcePartyListsWrapper}>
+      <NewPartyListModal
+        ref={createPartyListModalRef}
+        org={selectedContext}
+        env={env}
+        navigateUrl={`${getResourcePageURL(
+          selectedContext,
+          repo,
+          resourceData.identifier,
+          'partylists',
+        )}/${env}/`}
+        onClose={() => createPartyListModalRef.current?.close()}
```

Comment:
Can we add a test for this?

Reply:
For this to be tested, we would have to trigger modal close from the child component, which means the parent component being tested must know how to trigger the closing of the child component. I think we should consider how we would like to test this for all new modals, when we replace modal with new component from design system.